crypto/kyber/p_kyber.c

```diff
@@ -29,6 +29,53 @@ static int pkey_kyber512_keygen(EVP_PKEY_CTX *ctx, EVP_PKEY *pkey) {
   return 1;
 }
 
+static int pkey_kyber512_encapsulate(EVP_PKEY_CTX *ctx,
+                                     uint8_t *ciphertext,
+                                     size_t  *ciphertext_len,
+                                     uint8_t *shared_secret,
+                                     size_t  *shared_secret_len) {
+  *ciphertext_len    = KYBER512_CIPHERTEXT_BYTES;
+  *shared_secret_len = KYBER512_SHARED_SECRET_BYTES;
+  if (ciphertext == NULL) { // Caller is performing a size check
+    return 1;
+  }
+
+  if (ctx != NULL &&
+      ctx->pkey != NULL &&
+      ctx->pkey->pkey.ptr != NULL &&
+      ctx->pkey->type == EVP_PKEY_KYBER512) {
```

Comment:
I think we should try and branch on errors, not success. Unless it totally does not make sense.